compiler/p/codegen/OMRLoadStoreHandler.hpp

```diff
@@ -0,0 +1,80 @@
+/*******************************************************************************
+ * Copyright (c) 2020, 2020 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#ifndef OMR_POWER_LOADSTOREHANDLER_INCL
+#define OMR_POWER_LOADSTOREHANDLER_INCL
+
+#include "codegen/InstOpCode.hpp"
+#include "codegen/MemoryReference.hpp"
+#include "codegen/Register.hpp"
+#include "il/Node.hpp"
+#include "infra/Annotations.hpp"
+
+namespace OMR
+{
+
+namespace Power
+{
+class NodeMemoryReference
+    {
+    TR::MemoryReference *_memRef;
+
+    public:
+    NodeMemoryReference(TR::MemoryReference *memRef) : _memRef(memRef) {}
+
+    TR::MemoryReference *getMemoryReference() { return _memRef; }
+    void decReferenceCounts(TR::CodeGenerator *cg) { return _memRef->decNodeReferenceCounts(cg); }
+    };
```

Comment:
In #5630 it is noted this class is going to take care of node counts. However it seems all it does not is delegate this to the memory reference. Has that code not been extracted into this class yet or is this the final implementation?

Reply:
This code simply hasn't been extracted out yet. It can't be moved out of `TR::MemoryReference` until the old API is removed, but that's still being relied upon by OpenJ9 at the moment.

Once the new API is consistently utilized, the only remaining direct users of `TR::MemoryReference`'s node tracking capabilities should be this class itself and the `generateMemoryReference` function. At that time, it should be trivial to move the code for tracking node references and decrementing them at the appropriate time into this class.